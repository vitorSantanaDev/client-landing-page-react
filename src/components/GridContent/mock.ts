export default {
  title: 'Title text example',
  content: `<p>Lorem ipsum dolor sit amet, <a href="#">consectetur adipiscing</a> elit. Sed ac lacus pretium urna malesuada dignissim. Sed risus tortor, sagittis eget sem ut, porta imperdiet magna. In fringilla nisi enim. Aenean mollis vehicula nulla, non vulputate eros pretium eget. Vivamus tincidunt accumsan massa non convallis. Aliquam molestie efficitur faucibus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Vestibulum suscipit convallis mi, vitae ornare dolor hendrerit quis. Nam gravida orci posuere nisi imperdiet posuere. Phasellus quis blandit turpis, eu feugiat velit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Integer ac malesuada quam. Aliquam eu finibus massa.</p>

  <p>Donec pulvinar volutpat ante quis porttitor. In hac habitasse platea dictumst. Sed eu placerat turpis, vitae pulvinar odio. Pellentesque viverra posuere enim, eu dapibus quam congue sit amet. Sed iaculis eleifend scelerisque. Curabitur ut elit at sem cursus euismod. Praesent ornare, urna vitae sollicitudin tincidunt, sem odio accumsan velit, quis iaculis neque elit vel ante. Suspendisse rhoncus felis at arcu blandit facilisis. Cras euismod varius nibh, a cursus nisi. Integer laoreet posuere sem vel pulvinar. Etiam vel blandit libero. Praesent consectetur tellus et justo lobortis rhoncus.</p>
  `,
  background: false
}
