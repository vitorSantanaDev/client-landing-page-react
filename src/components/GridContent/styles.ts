import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`
export const Content = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`
