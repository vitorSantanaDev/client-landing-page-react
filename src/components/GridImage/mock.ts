export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      image: {
        altText: 'Teste 1',
        srcImage: 'https://source.unsplash.com/random/800x800?r=1'
      }
    },
    {
      image: {
        altText: 'Teste 2',
        srcImage: 'https://source.unsplash.com/random/800x800?r=2'
      }
    },
    {
      image: {
        altText: 'Teste 3',
        srcImage: 'https://source.unsplash.com/random/800x800?r=3'
      }
    },
    {
      image: {
        altText: 'Teste 4',
        srcImage: 'https://source.unsplash.com/random/800x800?r=4'
      }
    },
    {
      image: {
        altText: 'Teste 5',
        srcImage: 'https://source.unsplash.com/random/800x800?r=5'
      }
    },
    {
      image: {
        altText: 'Teste 6',
        srcImage: 'https://source.unsplash.com/random/800x800?r=6'
      }
    }
  ]
}
