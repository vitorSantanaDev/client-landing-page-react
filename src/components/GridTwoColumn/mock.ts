export default {
  title: 'Grid two column',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ad vero, accusantium eum exercitationem in ex quo veniam sed totam, quos ipsam at. Facilis a labore alias sed, natus dolor temporibus?',
  srcImage: 'assets/images/javascript.svg'
}
