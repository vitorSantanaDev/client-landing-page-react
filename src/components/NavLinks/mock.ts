export default [
  { children: 'First Link', link: '#target_1', newTab: false },
  { children: 'Second Link', link: '#target_2', newTab: true },
  { children: 'Tertiary Link', link: '#target_3', newTab: false },
  { children: 'Last Link', link: '#target_4', newTab: true },
  { children: 'Test Link', link: '#target_5', newTab: false },
  { children: 'Contact Link', link: '#target_6', newTab: false },
  { children: 'About Link', link: '#target_7', newTab: true }
]
