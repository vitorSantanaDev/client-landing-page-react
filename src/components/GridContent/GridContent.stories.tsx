import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import GridContent from '.'
import mock from './mock'

export default {
  title: 'GridContent',
  component: GridContent,
  args: { ...mock }
} as ComponentMeta<typeof GridContent>

export const Default: ComponentStory<typeof GridContent> = (args) => (
  <GridContent {...args} />
)
