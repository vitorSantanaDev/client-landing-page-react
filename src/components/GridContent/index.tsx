import React from 'react'

import Heading from '../Heading'
import SectionBackground from '../SectionBackground'
import TextComponent from '../TextComponent'
import * as S from './styles'

interface IGridContent {
  title: string
  content: string
  background?: boolean
  sectionId?: string
}

const GridContent: React.FC<IGridContent> = ({
  title,
  content,
  background,
  sectionId
}) => (
  <SectionBackground sectionId={sectionId} background={background}>
    <S.Wrapper>
      <Heading uppercase colorDark={!background} as="h2">
        {title}
      </Heading>
      <S.Content>
        <TextComponent>{content}</TextComponent>
      </S.Content>
    </S.Wrapper>
  </SectionBackground>
)

export default GridContent
