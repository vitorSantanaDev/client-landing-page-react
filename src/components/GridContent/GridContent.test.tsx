import React from 'react'
import { renderTheme } from '../../styles/theme/render-theme'

import GridContent from '.'
import mock from './mock'

describe('<GridContent />', () => {
  it('should ', () => {
    const { container } = renderTheme(<GridContent {...mock} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
