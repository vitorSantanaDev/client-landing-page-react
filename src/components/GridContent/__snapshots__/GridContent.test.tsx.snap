// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContent /> should  1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem 0;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        Title text example
      </h2>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <p>
            Lorem ipsum dolor sit amet, 
            <a
              href="#"
            >
              consectetur adipiscing
            </a>
             elit. Sed ac lacus pretium urna malesuada dignissim. Sed risus tortor, sagittis eget sem ut, porta imperdiet magna. In fringilla nisi enim. Aenean mollis vehicula nulla, non vulputate eros pretium eget. Vivamus tincidunt accumsan massa non convallis. Aliquam molestie efficitur faucibus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Vestibulum suscipit convallis mi, vitae ornare dolor hendrerit quis. Nam gravida orci posuere nisi imperdiet posuere. Phasellus quis blandit turpis, eu feugiat velit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Integer ac malesuada quam. Aliquam eu finibus massa.
          </p>
          

  
          <p>
            Donec pulvinar volutpat ante quis porttitor. In hac habitasse platea dictumst. Sed eu placerat turpis, vitae pulvinar odio. Pellentesque viverra posuere enim, eu dapibus quam congue sit amet. Sed iaculis eleifend scelerisque. Curabitur ut elit at sem cursus euismod. Praesent ornare, urna vitae sollicitudin tincidunt, sem odio accumsan velit, quis iaculis neque elit vel ante. Suspendisse rhoncus felis at arcu blandit facilisis. Cras euismod varius nibh, a cursus nisi. Integer laoreet posuere sem vel pulvinar. Etiam vel blandit libero. Praesent consectetur tellus et justo lobortis rhoncus.
          </p>
          
  
        </div>
      </div>
    </div>
  </div>
</div>
`;
